### PR TITLE
Check functionTimeout value for empty string

### LINKED ISSUE #4332: Check for null when parsing functionTimeout from host.json
#### Investigative information
Same issue: https://github.com/Azure/azure-functions-host/issues/2468

Please provide the following:

- Timestamp: 04-18-2019 1000
- Function App version (1.0 or 2.0): 2x
- Function App name: Any
- Function name(s) (as appropriate): Any
- Invocation ID:
- Region:

#### Repro steps

Provide the steps required to reproduce the problem:

Set below attribute in host.json
"functionTimeout": null,

#### Expected behavior

Provide a description of the expected behavior.

Based on code, it should fail at  "if (value != null)" condition and set default timeout.
Based on 1x behavior, have unlimited timeout.
Documentation should be clear on the behavior.  Seems it was not addressed in https://github.com/Azure/azure-functions-host/issues/3603

#### Actual behavior

Provide a description of the actual behavior observed.

Gives start error: 
String was not recognized as a valid TimeSpan.

#### Known workarounds

Provide a description of any known workarounds.

Avoid setting null

### PR DESCRIPTION
**Pull request by jitinjohn**
When host.json has setting as "functionTimeout": null, it is translated to empty string as value here. 
Ideally, the parser should return null.

Fixes #4332

### CODE REVIEW

File: src/WebJobs.Script/Config/ScriptHostOptionsSetup.cs [hunk 1]
```diff
@@ -77,7 +77,7 @@ private void ConfigureFunctionTimeout(IConfigurationSection jobHostSection, Scri
             {
                 TimeSpan functionTimeout = _environment.IsDynamic() ? DefaultFunctionTimeoutDynamic : DefaultFunctionTimeout;
                 string value = jobHostSection.GetValue<string>("functionTimeout");
-                if (value != null)
+                if (!string.IsNullOrEmpty(value))
```
Comment (pragnagopa) [inline src/WebJobs.Script/Config/ScriptHostOptionsSetup.cs]: Add unit test: https://github.com/Azure/azure-functions-host/blob/dev/test/WebJobs.Script.Tests/Configuration/ScriptHostOptionsSetupTests.cs

**Comment by jitinjohn** [comment]
Abandoning in favor of a better fix. Refer PR #4371

**Comment by jitinjohn** [comment]
Turns out the valid fix is in this part of the code

**Comment by pragnagopa** [comment]
Build is still failing.

**Pull request merged**